plugin/src/main/java/io/jenkins/plugins/analysis/core/util/AffectedFilesResolver.java

```diff
@@ -308,7 +308,14 @@ CopyResult copyAllInBatch(final Report report, final FilteredLog log)
             if (workspace.getChannel() == null) {
                 throw new IOException("No channel available for batch copy");
             }
-            return workspace.act(new BatchFileCopier(report, permittedAbsolutePaths, buildFolder, log));
+            // Check which files already exist in build folder on controller to avoid security violation
+            Set<String> filesToSkip = new java.util.HashSet<>();
+            for (Issue issue : report) {
+                if (buildFolder.child(getZipName(issue.getFileName())).exists()) {
+                    filesToSkip.add(issue.getFileName());
+                }
+            }
```

Comment:
A comment `//` indicates that the following code should be extracted into a method.